### PR TITLE
Issue 4 update layout with new data

### PR DESCRIPTION
**Pull request by KotiJocki**
Added a test-case demonstrating that changing the text on a label from new test-data does not cause the label to be resized for the layout tests.

Added a fix to force re-laying out for each piece of data.

### CODE REVIEW

File: LayoutTestTests/LayoutTestCaseTests/LayoutTestCaseMultipleDataOverlapTests.m [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+- (void)failTest:(NSString *)errorMessage view:(UIView *)view {
+    self.testFailures++;
+}
+
+#pragma mark - LYTViewProvider
+
++ (NSDictionary *)dataSpecForTest {
+    return @{
+             @"view": [UnitTestViews viewWithLongStringOverlappingLabel],
+             @"text": [[LYTDataValues alloc] initWithValues:@[@"X", @"A long string that will cause overlap"]]
```
Comment (plivesey) [inline LayoutTestTests/LayoutTestCaseTests/LayoutTestCaseMultipleDataOverlapTests.m]: Can you change this to LYTStringValues? It looks like I can still reproduce the bug and it makes it more consistent.

So, if I change this to string values:
- Test passes
- If I remove layoutIfNeeded, test fails

File: LayoutTestTests/Helpers/UnitTestViews.m [hunk 2]
```diff
@@ -151,6 +152,31 @@ + (UIView *)viewWithAccessibilityIDButNoLabel {
...
+    label.translatesAutoresizingMaskIntoConstraints = NO;
+    label.text = @"X";
+    
+    view2.translatesAutoresizingMaskIntoConstraints = NO;
+    
+    [superview addSubview:label];
+    [superview addSubview:view2];
+    superview.label = label;
+    superview.translatesAutoresizingMaskIntoConstraints = NO;
+    [superview addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"H:|-0-[label]" options:NSLayoutFormatAlignAllTop metrics:nil views:viewsDictionary]];
```
Comment (plivesey) [inline LayoutTestTests/Helpers/UnitTestViews.m]: Can you add a small comment here describing what these contraints do and why it causes overlap for long strings?

**Comment by KotiJocki** [comment]
1) As I understand it, the size of a label is determined by its contents with intrinsicContentSize. Clearly setting the text on the label, although it changes the size of the view when rendered, does not cause the layout of the super-view to be updated. This might be a bug in iOS or it might be because these views are isolated and not really being rendered in a complete view hierarchy, or it might be for performance reasons. 

There is an argument that it should be the responsibility of the user to call setNeedsLayout in the viewForData method when they make a change that will change a views intrinsic content size, but I feel this would add greater complexity for the user.

2) Going by http://semver.org/ I would say this was a patch version increment.

**Comment by plivesey** [comment]
Yeah, that makes sense. Effectively, setting text on a label doesn't invalidate layout, though it may cause the view to layout differently if you force it. I definitely don't think you should leave it up to the user to call this.

**Comment by plivesey** [comment]
For 2, yeah, after thinking about it, I think this should be a patch upgrade.

**Comment by plivesey** [comment]
Ok, this lgtm.
I'll do a release with a patch increase soon.

**Comment by plivesey** [comment]
Merged. Thanks a lot for the bug fix and unit tests.

**Pull request merged**